### PR TITLE
Tests - Refactoring puppeteer tests

### PR DESCRIPTION
**Pull request by boubkerbribri**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Refactoring puppeteer tests to be much easier to read
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | sanity tests on travis should pass

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/16194)

### CODE REVIEW

File: tests/puppeteer/campaigns/sanity/05_cartFO/01_editCheckCart.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ const init = async function () {
   Check the cart
   Edit the cart and check it
  */
-describe('Check the Product page', async () => {
+describe('Check Cart in FO', async () => {
```
Comment (SimonGrn) [inline tests/puppeteer/campaigns/sanity/05_cartFO/01_editCheckCart.js:33]: nice catch

File: tests/puppeteer/pages/BO/addCategory.js [hunk 2]
```diff
@@ -37,11 +37,9 @@ module.exports = class AddCategory extends BOBasePage {
     await this.setValue(this.metaTitleInput, categoryData.metaTitle);
     await this.setValue(this.metaDescriptionTextarea, categoryData.metaDescription);
     await this.page.click(this.selectAllGroupAccessCheckbox);
-    await Promise.all([
-      this.page.waitForNavigation({waitUntil: 'networkidle0'}),
-      this.page.waitForSelector(this.alertSuccessBlockParagraph, {visible: true}),
-      this.page.click(this.saveCategoryButton),
-    ]);
+    // Save Category
+    await this.clickAndWaitForNavigation(this.saveCategoryButton);
```
Comment (SimonGrn) [inline tests/puppeteer/pages/BO/addCategory.js:41]: What was the problem here ?

**Review by SimonGrn — DISMISSED** [review]

File: tests/puppeteer/pages/BO/addCategory.js (hunk 2 above)
Comment (boubkerbribri) [reply tests/puppeteer/pages/BO/addCategory.js:41]: No problem here, just replacing same code (`Promise.all click and wait for navigation`) with function that do the same

**Review by matthieu-rolland — APPROVED** [review]
LGTM

**Pull request merged**